src/core/read-stream.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * @private
+ * @class
+ * @name ReadStream
+ * @classdesc Helper class for organized reading of memory.
+ */
+class ReadStream {
+    constructor(arraybuffer) {
+        this.arraybuffer = arraybuffer;
+        this.dataView = new DataView(arraybuffer);
+        this.offset = 0;
+        this.stack = [];
+    }
+
+    get remainingBytes() {
+        return this.dataView.byteLength - this.offset;
+    }
+
+    reset(offset = 0) {
+        this.offset = offset;
+    }
+
+    skip(bytes) {
+        this.offset += bytes;
+    }
+
+    align(bytes) {
+        this.offset = (this.offset + bytes - 1) & (~(bytes - 1));
+    }
+
+    _inc(amount) {
+        this.offset += amount;
+        return this.offset - amount;
+    }
```

Comment:
not the cleanest implementation and its usage bellow :)

Reply:
I don't love it either, but how would you suggest improving it?